### PR TITLE
Refactor 5-app-infra-step

### PR DESCRIPTION
**Pull request by mariammartins**
Hey folks, this PR contains:

- 5-app-infra refactoring step
- Update of steps to deploy 5-app-infra locally
- Changing step 6-machine-learning to "vertex-ai-machine-learning-pipeline" as an example

### CODE REVIEW

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,1003 @@
+# Vertex-ai-machine-learning-pipeline
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:1]: adjust title

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,1003 @@
...
+
+After executing this stage, unset the `GOOGLE_IMPERSONATE_SERVICE_ACCOUNT` environment variable.
+
+  ```bash
+  unset GOOGLE_IMPERSONATE_SERVICE_ACCOUNT
+  ```
+
+## Post Deployment
+
+### BIG QUERY
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:568]:
```suggestion
### Big Query
```

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 3]
```diff
@@ -0,0 +1,1003 @@
...
+      "operations" = {
+        "cloudbuild.googleapis.com" = {
+          "methods" = ["*"]
+        }
+      }
+    }
+  },
+```
+
+### MACHINE LEARNING PIPELINE
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:831]:
```suggestion
### MACHINE LEARNING PIPELINE
```
```suggestion
### MACHINE LEARNING PIPELINE
```
```suggestion
### Machine Learning Pipeline
```

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 4]
```diff
@@ -0,0 +1,1003 @@
...
+     args: ['cp', './common/vertex-ai-pipeline/pipeline_package.yaml', 'gs://{your-composer-bucket}/dags/common/vertex-ai-pipeline/']
+     id: 'upload_composer_file'
+
+ # upload pipeline dag to composer
+    - name: 'gcr.io/cloud-builders/gsutil'
+      args: ['cp', './composer/dags/dag.py', 'gs://{your-composer-bucket}/dags/']
+      id: 'upload dag'
+```
+
+### 3. Configure variables in compile_pipeline.py and runpipeline.py
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:924]:
```suggestion
#### 3. Configure variables in compile_pipeline.py and runpipeline.py
```

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 5]
```diff
@@ -0,0 +1,1003 @@
...
+    |deployment_config['encryption']|The kms key for the prod env. This key is used to encrypt the vertex model, endpoint, model deployment, and model monitoring.|`projects/{prod-kms-project}/locations/us-central1/keyRings/{keyring-name}/cryptoKeys/{key-name}`|Navigate to `fldr-production/prj-n-kms`, navigate to the Security/Key management in that project to find the key in `sample-keyring` keyring of your target region `us-central1`|
+    |encryption_spec_key_name|The name of the encryption key for the non-prod env. This key is used to create the vertex pipeline job and it's associated metadata store|`projects/{non-prod-kms-project}/locations/us-central1/keyRings/{keyring-name}/cryptoKeys/{key-name}`|Navigate to `fldr-non-production/prj-n-kms`, navigate to the Security/Key management in that project to find the key in `sample-keyring` keyring of your target region `us-central1`|
+    |monitoring_config['email']|The email that Vertex AI monitoring will email alerts to|`your email`|your email associated with your gcp account|
+
+The compile_pipeline.py and runpipeline.py files are commented to point out these variables.
+
+### 4. Merge and deploy
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:944]:
```suggestion
#### 4. Merge and deploy
```

File: examples/vertex-ai-machine-learning-pipeline/README.md [hunk 6]
```diff
@@ -0,0 +1,1003 @@
...
+
+- Once everything is configured, you can commit your changes and push to the dev branch. Then, create a PR to from dev to staging(non-prod) which will result in triggering the pipeline if approved. The vertex pipeline takes about 30 minutes to finish and if there are no errors, a trained model will be deployed to and endpoint in the prod project which you can use to make prediction requests.
+
+### 5. Model Validation
```
Comment (caetano-colin) [inline examples/vertex-ai-machine-learning-pipeline/README.md:948]:
```suggestion
#### 5. Model Validation
```

**Review by caetano-colin — CHANGES_REQUESTED** [review]
Some style changes

**Comment by caetano-colin** [comment]
Some things to be done before merge:

- Adjust main README.md on section [6. ml-pipeline](https://github.com/mariammartins/terraform-google-enterprise-genai/blob/refactor/5-app-infra-step/6-ml-pipeline). Remove the section and add on readme a section about the examples, and link the example there.

**Comment by sleighton2022** [comment]
Linting fails

File: README.md [hunk 7]
```diff
@@ -23,7 +23,7 @@ Alternatively, the user can follow steps on [`docs/deploy_on_foundation_v4.0.0.m
 
 Additional steps were added to provide an example Machine Learning application:
 
-* [6. ml-pipeline](./6-ml-pipeline/)
+* [Machine-learning-pipeline](./examples/machine-learning-pipeline)
```
Comment (caetano-colin) [inline README.md:26]:
```suggestion
* [machine-learning-pipeline](./examples/machine-learning-pipeline)
```

**Comment by mariammartins** [comment]
> Linting fails

Fixed

File: examples/machine-learning-pipeline/README.md [hunk 8]
```diff
@@ -0,0 +1,1003 @@
+# Machile Learning Pipeline Overview
```
Comment (renato-rudnicki) [inline examples/machine-learning-pipeline/README.md:1]: Machile Learning Pipeline Overview
```suggestion
# Machine Learning Pipeline Overview
```

Comment (renato-rudnicki) [reply examples/machine-learning-pipeline/README.md:1]: Just a small typo.

**Pull request merged**